src/mpid/ch4/shm/xpmem/xpmem_control.h

```diff
@@ -9,10 +9,15 @@
 
 #include "shm_types.h"
 
-int MPIDI_XPMEM_ctrl_send_lmt_rts_cb(MPIDI_SHM_ctrl_hdr_t * ctrl_hdr);
-int MPIDI_XPMEM_ctrl_send_lmt_cts_cb(MPIDI_SHM_ctrl_hdr_t * ctrl_hdr);
-int MPIDI_XPMEM_ctrl_send_lmt_send_fin_cb(MPIDI_SHM_ctrl_hdr_t * ctrl_hdr);
-int MPIDI_XPMEM_ctrl_send_lmt_recv_fin_cb(MPIDI_SHM_ctrl_hdr_t * ctrl_hdr);
-int MPIDI_XPMEM_ctrl_send_lmt_cnt_free_cb(MPIDI_SHM_ctrl_hdr_t * ctrl_hdr);
+int MPIDI_XPMEM_rts_cb(int handler_id, void *am_hdr, void **data, size_t * data_sz, int is_local,
+                       int *is_contig, MPIR_Request ** req);
+int MPIDI_XPMEM_cts_cb(int handler_id, void *am_hdr, void **data, size_t * data_sz, int is_local,
+                       int *is_contig, MPIR_Request ** req);
+int MPIDI_XPMEM_send_fin_cb(int handler_id, void *am_hdr, void **data, size_t * data_sz,
+                            int is_local, int *is_contig, MPIR_Request ** req);
+int MPIDI_XPMEM_recv_fin_cb(int handler_id, void *am_hdr, void **data, size_t * data_sz,
+                            int is_local, int *is_contig, MPIR_Request ** req);
+int MPIDI_XPMEM_cnt_free_cb(int handler_id, void *am_hdr, void **data, size_t * data_sz,
+                            int is_local, int *is_contig, MPIR_Request ** req);
 
```

Comment:
I'd prefer the original function names. In future we may add callback for non-LMT messages, or receiver-initialized LMT. So we need the `send_lmt` prefix.

Reply:
I was wondering that may be the plan. 

Can we wait until the new components are on the table, then we can look at the actual objects and decide a good naming scheme. Otherwise, we are having these long names (which obviously makes formatting and reading more difficult) for no benefit, right?

Reply:
Okay, we will add you to the reviewer. The PR won't merge until all the approvals :) You'll have time to get ready

Reply:
> We have our own shmmods and following through this format. 

Just for my understanding, you guys have shmmods similar to xpmem that are not really standalone shmmods, but components implements posix shmmod message protocols such as large message transfer, right?

Reply:
Yes, we are using SHM am ctrl message. We take the design from upstream xpmem mod and make use of the infrastructure that's there. I'm trying to have our code structure consistent with other shmmod (xpmem). 

But since we have a separated mod, the merge is not gonna be difficult. It's just that we will lost this consistency. Which is fine. It's just a bummer. Note that we are targeting 3.4a2, so this won't affect us until the next merge.

I don't have anything against this PR. I think the new name actually make more sense. I just want to voice my concern about the consistency, because we are at the end of the stream. Just a minor nitpick . XD